lib/oscap_tests.pm

```diff
@@ -41,53 +59,367 @@ our $f_vlevel = 'ERROR';
+
+our $ansible_playbook_standart = "opensuse-playbook-standard.yml";
+
 # The OS status of remediation: '0', not remediated; '>=1', remediated
 our $remediated = 0;
 
-# Evaluate execution count
-our $evaluate_count = 3;
+# Is it ansible remediation: '0', bash remediation; '1' ansible remediation
+our $ansible_remediation = 0;
+
+# Directory where ansible playbooks resides on local system
+our $ansible_file_path = "/usr/share/scap-security-guide/ansible/";
+our $full_ansible_file_path = "";
+
+# Variables $use_content_type and $remove_rules_missing_fixes are fetched from configuration file located in:
+#https://gitlab.suse.de/seccert-public/compliance-as-code-compiled/-/blob/main/content/openqa_config.conf
+# Configuration for the contenyt type (source of ds, ansible and xccdf files):
+# If set to 1 - tests will use files from scap-security-guide package
+# If set to 2 - tests will use files from compliance-as-code-compiled repository - https://gitlab.suse.de/seccert-public/compliance-as-code-compiled
+# If set to 3 - tests will use files cloned and built from ComplianceAsCode repository master branch - https://github.com/ComplianceAsCode/content.git
+our $use_content_type = 1;
+
+# Option configures to use or not functionality to remove from DS and ansible file rules for which do not have remediations
+# If set to 1 - rules for which do not have remediations will be removed from DS and ansible file rules for which do not have remediations
+# If set to 0 - no changes done.
+our $remove_rules_missing_fixes = 1;
+
+# Option configures to use or not functionality to exclude rules for profiles defined in file:
+# https://gitlab.suse.de/seccert-public/compliance-as-code-compiled/-/blob/main/content/openqa_tests_exclusions.yaml
+# If set to 1 - rules defined in exclusions files are excluded for remediation and evaluation
+# If set to 0 - rules defined in exclusions files are not used
+our $use_exclusions = 1;
+
+# Keeps count of reboots to control it
+our $reboot_count = 0;
+
+# evaluate execution count done by test execution. Set in oscap_security_guide_setup.pm
+# and "security/oscap_stig/oscap_xccdf_eval" need to be set in the schedule yaml file accordingly
+our $evaluate_count = 2;
+
+# Stores CCE IDs of failed ansible remediation tasks.
+# Used in second ansible remediation as exclusions.
+our $failed_cce_ids_ref;
+
+# List to collect needed run results
+our @test_run_report = ();
+our $test_run_report_name = "test_run_report.txt";
+
+# Get sle version "sle12" or "sle15"
+our $sle_version = '';
 
 # Upload HTML report by default
 set_var('UPLOAD_REPORT_HTML', 1);
 
 # Set value for 'scap-security-guide' ds file
 sub set_ds_file {
-
     # Set the ds file for separate product, e.g.,
     # for SLE15 the ds file is "ssg-sle15-ds.xml";
     # for SLE12 the ds file is "ssg-sle12-ds.xml";
     # for Tumbleweed the ds file is "ssg-opensuse-ds.xml"
     my $version = get_required_var('VERSION') =~ s/([0-9]+).*/$1/r;
```

Comment:
instead of running this regex in multiple places looks like these two methods could be replaced with openQA variables and be fixed there as well.